stratum/hal/lib/common/phaldb_service_test.cc

```diff
@@ -0,0 +1,861 @@
+#include "google/rpc/code.pb.h"
+#include "stratum/glue/net_util/ports.h"
+// Note: EXPECT_OK already defined in google protobuf status.h
+#undef EXPECT_OK
+#include "stratum/glue/status/status_test_util.h"
+#include "stratum/hal/lib/common/error_buffer.h"
+#include "stratum/hal/lib/common/phal_mock.h"
+#include "stratum/hal/lib/phal/attribute_database_mock.h"
+#include "stratum/lib/security/auth_policy_checker_mock.h"
+#include "stratum/lib/test_utils/matchers.h"
+#include "stratum/lib/utils.h"
+#include "stratum/lib/macros.h"
+#include "stratum/public/lib/error.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "stratum/glue/integral_types.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/substitute.h"
+
+namespace stratum {
+namespace hal {
+
+using ::testing::_;
+using ::testing::Eq;
+using ::testing::DoAll;
+using ::testing::HasSubstr;
+using ::testing::Invoke;
+using ::testing::Return;
+using ::testing::SaveArg;
+using ::testing::ByMove;
+using ::testing::SetArgPointee;
+using ::testing::WithArgs;
+
+typedef ::grpc::ClientReader<::stratum::hal::phal::SubscribeResponse>
+    ClientStreamChannelReader;
+
+class PhalDBServiceTest : public ::testing::TestWithParam<OperationMode> {
+ protected:
+  void SetUp() override {
+    mode_ = GetParam();
+    phal_mock_ = absl::make_unique<PhalMock>();
+    auth_policy_checker_mock_ = absl::make_unique<AuthPolicyCheckerMock>();
+    error_buffer_ = absl::make_unique<ErrorBuffer>();
+    phaldb_service_ = absl::make_unique<PhalDBService>(mode_,
+                                               phal_mock_.get(),
+                                               auth_policy_checker_mock_.get(),
+                                               error_buffer_.get());
+    std::string url =
+        "localhost:" + std::to_string(stratum::PickUnusedPortOrDie());
+    ::grpc::ServerBuilder builder;
+    builder.AddListeningPort(url, ::grpc::InsecureServerCredentials());
+    builder.RegisterService(phaldb_service_.get());
+    server_ = builder.BuildAndStart();
+    ASSERT_NE(server_, nullptr);
+    stub_ = ::stratum::hal::phal::PhalDBSvc::NewStub(
+        ::grpc::CreateChannel(url, ::grpc::InsecureChannelCredentials()));
+    ASSERT_NE(stub_, nullptr);
+
+    database_mock_ =
+        absl::make_unique<::stratum::hal::phal::AttributeDatabaseMock>();
+  }
+
+  void TearDown() override {
+    phaldb_service_->Teardown();
+    server_->Shutdown();
+  }
+
+  OperationMode mode_;
+  std::unique_ptr<AuthPolicyCheckerMock> auth_policy_checker_mock_;
+  std::unique_ptr<ErrorBuffer> error_buffer_;
+  std::unique_ptr<PhalDBService> phaldb_service_;
+  std::unique_ptr<::grpc::Server> server_;
+  std::unique_ptr<::stratum::hal::phal::PhalDBSvc::Stub> stub_;
+  std::unique_ptr<PhalMock> phal_mock_;
+  std::unique_ptr<::stratum::hal::phal::AttributeDatabaseMock> database_mock_;
+};
+
+TEST_P(PhalDBServiceTest, SetupWarm) {
+  auto status = phaldb_service_->Setup(true);
+  EXPECT_TRUE(status.ok());
```

Comment:
Maybe `EXPECT_OK(phaldb_service_->Setup(true));` ?

Reply:
had some conflict with the  stratum/glue/status/status_test_util.h, have now removed that and EXPECT_OK now works